src/pyscipopt/scip.pxi

```diff
@@ -4621,6 +4621,31 @@ cdef class Model:
         if not self.getStage() >= SCIP_STAGE_SOLVING:
             raise Warning("method cannot be called before problem is solved")
         return self.getSolVal(self._bestSol, expr)
+    
+    def hasPrimalRay(self):
+        """
+        Returns whether a primal ray is stored that proves unboundedness of the LP relaxation
+        """
+        return SCIPhasPrimalRay(self._scip)
+        
+    def getPrimalRayVal(self, Variable var):
+        """
+        Gets value of given variable in primal ray causing unboundedness of the LP relaxation
+        """
+        return SCIPgetPrimalRayVal(self._scip, var.scip_var)
+
+    def getPrimalRay(self):
+        """
+        Gets primal ray causing unboundedness of the LP relaxation
+        """
+        cdef int _nvars = SCIPgetNVars(self._scip)
```

Comment:
We should probably check here if a primal ray exists, to avoid a segmentation fault. In case it doesn't exist we should throw an error that explains it.

Reply:
Ah, nice one, Mo! Added it now.